app/models/manageiq/providers/autosde/inventory/collector/target_collection.rb

```diff
@@ -8,49 +8,62 @@ def initialize(_manager, _target)
   def physical_storages
     return [] if references(:physical_storages).blank?
 
-    super
+    @physical_storages ||= references(:physical_storages).map do |ems_ref|
+      @manager.autosde_client.StorageSystemApi.storage_systems_get.map { |s| s if s.uuid == ems_ref }.compact.first
```

Comment:
This looks like it should be a `.select` rather than `map`.  Select does exactly what you're trying to do here with the `s if s.uuid == ems_ref` block.

```suggestion
      @manager.autosde_client.StorageSystemApi.storage_systems_get.select { |s| s.uuid == ems_ref }
```